src/adapters/DebugProtocolAdapter.ts

```diff
@@ -313,7 +313,19 @@ export class DebugProtocolAdapter {
                 this.emit('diagnostics', diagnostics);
             });
 
+            this.socketDebugger.on('control-connected', () => {
+                this.connectionDeferred.resolve();
+
+                this.logger.log(`Closing telnet connection used for compile errors`);
```

Comment:
Let's not close this. The compileClient (telnet connection) should stay alive for the lifetime of the debug session (i.e. until someone clicks "stop")

Reply:
This was a small step to start using the telnet logs to trigger the control port connection.

I'm starting the work to keep the debug session open once the control port closes.